api/server/middleware/user_agent.go

```diff
@@ -16,6 +17,7 @@ func NewUserAgentMiddleware(versionCheck string) Middleware {
 
 	return func(handler httputils.APIFunc) httputils.APIFunc {
 		return func(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+			ctx = dockerversion.WithUserAgent(ctx, r.Header.Get("User-Agent"))
```

Comment:
@calavera  When you say that middleware needs to be "docker packages free" are you saying that your desired end stat is for `api/server/middleware/user_agent.go`  to not import any packages other than the ones below?

```
 	"net/http"
 	"strings"
    	"github.com/Sirupsen/logrus"
 	"golang.org/x/net/context"
```